code-challenges.test.js

```diff
@@ -15,43 +18,111 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+describe("arrShuffle", () => {
+  it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    const result1 = arrShuffle(colors1)
+    expect(result1).toHaveLength(colors1.length - 1)
+    expect(colors1.slice(1)).toEqual(expect.arrayContaining(result1))
+
+    const colors2 = [
+      "chartreuse",
+      "indigo",
+      "periwinkle",
+      "ochre",
+      "aquamarine",
+      "saffron"
+    ]
+    const result2 = arrShuffle(colors2);
+    expect(result2).toHaveLength(colors2.length - 1);
+    expect(colors2.slice(1)).toEqual(expect.arrayContaining(result2))
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+const arrShuffle = (arr) => {
+  const shuffledArray = [...arr.slice(1)]
+
+  for (let i = shuffledArray.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1))
+    const temp = shuffledArray[i]
+    shuffledArray[i] = shuffledArray[j]
+    shuffledArray[j] = temp
+  }
+
+  return shuffledArray
+}
+
+
+
 // Pseudo code:
+// The function arrShuffle takes an array arr as input.
+// A new array shuffledArray is created by using the spread operator (...) on arr.slice(1). This creates a shallow copy of arr excluding the first element.
+// The for loop is used to iterate over the elements of shuffledArray in reverse order. It starts from the last index (shuffledArray.length - 1) and goes down to index 1.
+// Inside the loop, a variable j is declared and initialized with a random index between 0 and i (inclusive). The expression Math.floor(Math.random() * (i + 1)) generates a random number between 0 and i.
+// Next, a temporary variable temp is created and assigned the value of shuffledArray[i]. This is done to temporarily store the value at index i.
+// The elements at indices i and j in shuffledArray are swapped using array destructuring assignment. This is done by assigning shuffledArray[j] to shuffledArray[i] and vice versa.
+// The loop continues until all the elements have been shuffled.
```

Comment:
Good research on the Fisher Yates shuffle algorithm. Like I mentioned above, review your test as it currently is not testing that this function is working properly